integration/test_flow_integration.py

```diff
@@ -168,3 +168,39 @@ def test_write_to_tsdb():
     df = pd.DataFrame(expected, columns=columns)
     df.set_index(keys=['time'], inplace=True)
     assert res.equals(df), f"result{res}\n!=\nexpected{df}"
+
+
+def test_join_by_key(setup_kv_teardown_test):
+    table = Table(setup_kv_teardown_test, V3ioDriver())
```

Comment:
Can we have a unit test as well? Probably need to add a static data table for that.